### ISSUE TITLE
BSPLAYER always results in 100% match

### ISSUE THREAD

**Issue by zandadoum** [morpheus65535/bazarr#829]
Hello,

like the title sais, subs found through BSPLAYER always come back as a 100% match.
many times, they are out of sync or even the wrong episode, but they're always 100% matches.

this is specially horrible when using "Upgrade Previously Downloaded Subtitles" as a sub may be downloaded through BSPLAYER upgrading a 95% that was actually good with a bad one.

I am not sure if this is BSPLAYERs fault or if there is something that could be fixed through bazarr

**Comment by morpheus65535** [comment]
@GermanG Is there something wrong with this provider? It looks like it always match on hash...

**Comment by rigas40** [comment]
i confirm same problem

**Comment by GermanG** [comment]
TL;DR 
Answering your question: it's bsplayersubtitles issue and not Bazarr's.

Explained:
This issue is similar to #821 
Yes, bsplayer is a hash based provider only, and because of that the subliminal scoring system gives it a max score when it finds a subtitle. The fix pannal submitted might improve the experience.

Also because this provider is relaying heavily on the Kodi plugin (there's no documentation about bsplayersubtitles API except that code), you should experience the same problem there.

Regards

**Comment by morpheus65535** [comment]
@GermanG could you implement the missing code?

**Comment by GermanG** [comment]
@morpheus65535 yup, added to my ToDo. Maybe this weekend.
